### PR TITLE
ESQL: Better tests to AUTO_BUCKET

### PR DESCRIPTION
**Pull request by nik9000**
This improves the tests for AUTO_BUCKET marginally, specifically so that it tests all valid combinations of arguments and generates a correct types table. This'll combine nicely with #106782 to generate the signatures that kibana needs for it's editor.

### CODE REVIEW

**Review by bpintea — APPROVED** [review]
LGTM.
I was looking at renaming this function (#107197) and extend it to take just a span (#107272), temporal or numeric and the testing refactoring is welcome

File: x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/AutoBucketTests.java [hunk 1]
```diff
@@ -13,126 +13,143 @@
...
+            v = (long) value;
+        } else {
+            v = value;
+        }
+        return new TestCaseSupplier.TypedData(v, type, name).forceLiteral();
     }
 
-    private static Matcher<Object> numericResultsMatcher(List<TestCaseSupplier.TypedData> typedData, Object value) {
-        return equalTo(value);
+    private static TestCaseSupplier.TypedData keywordDateLiteral(String name, DataType type, String date) {
```
Comment (bpintea) [inline x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/AutoBucketTests.java:134]: Leftover?

Comment (nik9000) [reply x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/AutoBucketTests.java:134]: Checking

File: x-pack/plugin/esql/src/main/java/generated/org/elasticsearch/xpack/esql/expression/function/scalar/string/LocateNoStartEvaluator.java
Comment (fang-xing-esql) [inline x-pack/plugin/esql/src/main/java/generated/org/elasticsearch/xpack/esql/expression/function/scalar/string/LocateNoStartEvaluator.java:1]: Not relevant to this PR?

**Review by fang-xing-esql — APPROVED** [review]
LGTM! Except LocateNoStartEvaluator is not relevant.

File: x-pack/plugin/esql/src/main/java/generated/org/elasticsearch/xpack/esql/expression/function/scalar/string/LocateNoStartEvaluator.java
Comment (nik9000) [reply x-pack/plugin/esql/src/main/java/generated/org/elasticsearch/xpack/esql/expression/function/scalar/string/LocateNoStartEvaluator.java:1]: It's something we should have committed before. I figured I'd just land it with whatever PR. It looks like Alex landed it all on it's own though. So it's gone from this one.

**Pull request merged**